src/Sylius/Component/Core/spec/Promotion/Reverser/OrderItemPromotionAdjustmentsReverserSpec.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Component\Core\Promotion\Reverser;
+
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Core\Model\AdjustmentInterface;
+use Sylius\Component\Core\Model\Order;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\OrderItemUnitInterface;
+use Sylius\Component\Core\Promotion\Reverser\OrderItemPromotionAdjustmentsReverser;
+use Sylius\Component\Core\Promotion\Reverser\OrderItemPromotionAdjustmentsReverserInterface;
+use Sylius\Component\Promotion\Model\PromotionInterface;
+
+/**
+ * @author Gorka Laucirica <gorka.lauzirika@gmail.com>
+ */
+final class OrderItemPromotionAdjustmentsReverserSpec extends ObjectBehavior
+{
+    function it_is_initializable()
+    {
+        $this->shouldHaveType(OrderItemPromotionAdjustmentsReverser::class);
+    }
+
+    function it_implements_an_units_promotion_adjustments_reverser_interface()
+    {
+        $this->shouldImplement(OrderItemPromotionAdjustmentsReverserInterface::class);
+    }
+
+    function it_reverts_promotion_adjustments_on_all_units_of_given_order(
+        Order $order,
```

Comment:
We should use `OrderInterface` rather than `Order` class. Here and everywhere else ;)